File: src/component/modal/MessageModal.java
package component.modal;

import java.util.Arrays;

import org.newdawn.slick.Color;
import org.newdawn.slick.Font;
import org.newdawn.slick.gui.GUIContext;

import component.Label;
import component.Panel;
import component.Positionable;
import core.ConstantStore;
import core.FontStore;

public class MessageModal extends Modal {

	/**
	 * Constructs a {@code Modal} with a listener, message, and text for the dismiss button.
	 * @param context The GUI context
	 * @param listener The listener
	 * @param message The text for the message
	 */
	public MessageModal(GUIContext context, ModalListener listener, String message) {
		super(context, listener, message, 1);
		
		Font fieldFont = FontStore.get().getFont(FontStore.FontID.FIELD);
		this.messageLabel = new Label(container, DEFAULT_LABEL_WIDTH, fieldFont, Color.white, message);
		messageLabel.setAlignment(Label.Alignment.CENTER);
		
		int panelWidth = PADDING * 2 + messageLabel.getWidth();
		int panelHeight = PADDING * 3 + messageLabel.getHeight() + BUTTON_HEIGHT;
		panel = new Panel(container, panelWidth, panelHeight, ConstantStore.COLORS.get("MODAL"));
		panel.setBorderColor(ConstantStore.COLORS.get("MODAL_BORDER"));
		panel.setBorderWidth(2);
		panel.setLocation((getWidth() - panel.getWidth()) / 2, (getHeight() - panel.getHeight() / 2));
		
		createButtons();
		
		panel.add(messageLabel, panel.getPosition(Positionable.ReferencePoint.TOPCENTER), Positionable.ReferencePoint.TOPCENTER, 0, PADDING);
		
		panel.addAsRow(Arrays.asList(buttons).iterator(),
				panel.getPosition(Positionable.ReferencePoint.BOTTOMLEFT),
				PADDING,
				-PADDING - BUTTON_HEIGHT,
				PADDING);
		
		add(panel, getPosition(Positionable.ReferencePoint.CENTERCENTER), Positionable.ReferencePoint.CENTERCENTER);
	}
}

File: web/src/component/modal/MessageModal.ts
import { Modal, ModalListener } from './Modal';
import { Panel } from '../Panel';
import { Label, Alignment } from '../Label';
import { ReferencePoint } from '../Component';
import { Color } from '../../core/Color';
import { COLORS } from '../../core/ConstantStore';
import { FontStore, FontID } from '../../core/FontStore';

export class MessageModal extends Modal {
  constructor(width: number, height: number, listener: ModalListener, message: string) {
    super(width, height, listener, message, 1);

    const fieldFont = FontStore.getFont(FontID.FIELD);
    this.messageLabel = Label.withAutoHeight(Modal.DEFAULT_LABEL_WIDTH, fieldFont, Color.white, message);
    this.messageLabel.setAlignment(Alignment.CENTER);

    const panelWidth = Modal.PADDING * 2 + this.messageLabel.getWidth();
    const panelHeight = Modal.PADDING * 3 + this.messageLabel.getHeight() + Modal.BUTTON_HEIGHT;

    this.panel = new Panel(panelWidth, panelHeight, COLORS.get('MODAL') || Color.darkGray);
    this.panel.setBorderColor(COLORS.get('MODAL_BORDER') || Color.white);
    this.panel.setBorderWidth(2);
    this.panel.setLocation(
      Math.floor((this.getWidth() - this.panel.getWidth()) / 2),
      Math.floor((this.getHeight() - this.panel.getHeight()) / 2)
    );

    this.createButtons();

    this.panel.add(
      this.messageLabel,
      this.panel.getPosition(ReferencePoint.TOPCENTER),
      ReferencePoint.TOPCENTER,
      0,
      Modal.PADDING
    );

    this.panel.addAsRow(
      this.buttons,
      this.panel.getPosition(ReferencePoint.BOTTOMLEFT),
      Modal.PADDING,
      -Modal.PADDING - Modal.BUTTON_HEIGHT,
      Modal.PADDING
    );

    this.add(
      this.panel,
      this.getPosition(ReferencePoint.CENTERCENTER),
      ReferencePoint.CENTERCENTER
    );
  }
}
